src/libraries/Common/src/Interop/OSX/Swift.Runtime/CryptoKit.cs

```diff
@@ -0,0 +1,274 @@
+        /// </summary>
+        [StructLayout(LayoutKind.Sequential, Size = 16)]
+        internal unsafe partial struct SealedBox
+        {
+            internal Data combined;
+            internal readonly Data Ciphertext { get => CryptoKit.PInvoke_ChaChaPoly_SealedBox_GetCiphertext(this); }
+            internal readonly Data Tag { get => CryptoKit.PInvoke_ChaChaPoly_SealedBox_GetTag(this); }
+
+            internal SealedBox(ChaChaPoly.Nonce nonce, Data ciphertext, Data tag)
+            {
+                void* tagPtr = &tag;
+                void* ciphertextPtr = &ciphertext;
+                this = CryptoKit.PInvoke_ChaChaPoly_SealedBox_Init(nonce.payload, ciphertextPtr, tagPtr, ciphertext.Metadata(), tag.Metadata(), default(DataProtocol).WitnessTable(ciphertext), default(DataProtocol).WitnessTable(tag), out SwiftError error);
+
+                if (error.Value != null)
+                {
+                    throw new Exception();
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Represents AesGcm in C#.
+    /// </summary>
+    internal unsafe partial struct AesGcm
+    {
+        /// <summary>
+        /// Represents Nonce in C#.
+        /// </summary>
+        internal sealed unsafe partial class Nonce
+        {
+            private const int _payloadSize = 16;
+            internal void* payload;
+
+            internal Nonce()
+            {
+                payload = Marshal.AllocHGlobal(_payloadSize).ToPointer();
+                SwiftIndirectResult swiftIndirectResult = new SwiftIndirectResult(payload);
+                CryptoKit.PInvoke_AesGcm_Nonce_Init(swiftIndirectResult);
+            }
+
+            internal Nonce(Data data)
+            {
+                payload = Marshal.AllocHGlobal(_payloadSize).ToPointer();
+                SwiftIndirectResult swiftIndirectResult = new SwiftIndirectResult(payload);
+                void* dataPtr = &data;
+                CryptoKit.PInvoke_AesGcm_Nonce_Init2(swiftIndirectResult, dataPtr, data.Metadata(), default(DataProtocol).WitnessTable(data), out SwiftError error);
+
+                if (error.Value != null) // Should throw when nonce is not 12 bytes
+                {
+                    throw new Exception();
+                }
+            }
+
+            ~Nonce()
+            {
+                Marshal.FreeHGlobal(new IntPtr(payload));
+            }
+        }
+
+        /// <summary>
+        /// Represents SealedBox in C#.
+        /// </summary>
+        internal sealed unsafe partial class SealedBox
+        {
+            private const int _payloadSize = 24;
+            internal void* payload;
+            public Data Ciphertext { get => CryptoKit.PInvoke_AesGcm_SealedBox_GetCiphertext(new SwiftSelf(payload)); }
+            public Data Tag { get => CryptoKit.PInvoke_AesGcm_SealedBox_GetTag(new SwiftSelf(payload)); }
+            internal SealedBox(AesGcm.Nonce nonce, Data ciphertext, Data tag)
```

Comment:
There should be whitespace between constructors and the last property. This feedback applies elsewhere.